### PR TITLE
Compile with clang-11 on Linux and Windows

### PR DESCRIPTION
**Pull request by asvrada**
This PR includes the changes that enable locally building OpenEnclave using clang-11 on Linux and Windows.

CMake will now use clang-11 by default, this can be overridden with `cmake .. -DCMAKE_C_COMPILER=clang-10  -DCMAKE_CXX_COMPILER=clang++-10`

Will require changes to CI/pipeline to install clang-11 and set `CMAKE_C_COMPILER` flag for clang-10 when necessary

### CODE REVIEW

**Comment by anakrish** [comment]
bors try

**Comment by achamayou** [comment]
@asvrada thank you for making this change, it's hugely appreciated!

Unless I am missing something, I think there should be some changes when LVI mitigations are turned on. As I understand it, the purpose of the LVI_MITIGATION_BINDIR and running things from there is that clang-10 vanilla did not support this, but clang-11 does [1].

Extract from https://releases.llvm.org/11.0.0/tools/clang/docs/ClangCommandLineReference.html
```
-mlvi-cfi, -mno-lvi-cfi
Enable only control-flow mitigations for Load Value Injection (LVI)

-mlvi-hardening, -mno-lvi-hardening
Enable all mitigations for Load Value Injection (LVI)
```

It would be great to see this picked up and propagated like Spectre mitigation flags!

[1] https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/best-practices/optimized-mitigation-approach-load-value-injection.html
"Obtaining the mitigations tool"

**Comment by radhikaj** [comment]
> @asvrada thank you for making this change, it's hugely appreciated!
> 
> Unless I am missing something, I think there should be some changes when LVI mitigations are turned on. As I understand it, the purpose of the LVI_MITIGATION_BINDIR and running things from there is that clang-10 vanilla did not support this, but clang-11 does [1].
> 
> Extract from https://releases.llvm.org/11.0.0/tools/clang/docs/ClangCommandLineReference.html
> 
> ```
> -mlvi-cfi, -mno-lvi-cfi
> Enable only control-flow mitigations for Load Value Injection (LVI)
> 
> -mlvi-hardening, -mno-lvi-hardening
> Enable all mitigations for Load Value Injection (LVI)
> ```
> 
> It would be great to see this picked up and propagated like Spectre mitigation flags!
> 
> [1] https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/best-practices/optimized-mitigation-approach-load-value-injection.html "Obtaining the mitigations tool"

Hi @achamayou , we will have another PR for the LVI flow. There are two phases 1) This PR ensures we can build and run the SDK with clang-11. 2) The LVI flow needs tests for performance and inspection to ensure mitigations are properly applied. That will be another PR.

File: 3rdparty/compiler-rt/CMakeLists.txt [hunk 1]
```diff
@@ -5,9 +5,9 @@ include(FetchContent)
 FetchContent_Declare(
   compiler-rt-sources
   SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/compiler-rt
-  URL https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.1/compiler-rt-10.0.1.src.tar.xz
+  URL https://github.com/llvm/llvm-project/releases/download/llvmorg-11.1.0/compiler-rt-11.1.0.src.tar.xz
```
Comment (anakrish) [inline 3rdparty/compiler-rt/CMakeLists.txt:8]: Does this also work with previous clang-10?

**Review by anakrish — COMMENTED** [review]
LGTM.

Does this PR ensure that the SDK also works with clang-10?

File: 3rdparty/compiler-rt/CMakeLists.txt (hunk 1 above)
Comment (asvrada) [reply 3rdparty/compiler-rt/CMakeLists.txt:8]: The tests all pass when building with clang-10, so I think using v11.1.0 here works with clang-10.

**Comment by asvrada** [comment]
> ## try
> Build failed:
> 
> * [continuous-integration/jenkins/branch](https://openenclaveci.westus.cloudapp.azure.com/job/Bors/job/openenclave/job/trying/163/display/redirect)

* Azure-Linux failed because of code coverage, CI hasn't been updated (should be addressed in a separate PR) to use clang-11 to generate code coverage. With this PR, only clang-11 is supported to run code coverage.

* Agnostic-Linux failed because CI hasn't been updated to install clang-11

* Azure-Windows failed: to fix in this PR

**Comment by asvrada** [comment]
bors try

**Comment by achamayou** [comment]
@radhikaj fantastic, sorry for the unnecessary comment then! If you would like some feedback on performance regressions, I would be very happy to run the CCF performance tests on experimental builds.

**Comment by CyanDevs** [comment]
This is for a separate working branch called `clang-11` so go ahead and merge whenever this is ready.

**Review by anakrish — APPROVED** [review]
LGTM.

We do a deeper review when the PR is submitted for the main branch.

**Review by CyanDevs — APPROVED** [review]

**Comment by asvrada** [comment]
> Just to check: if `clang-11` and `clang` is specified for `find_program()`, it finds `clang-10` as well since `clang` is linked to `clang-10` in our CI nodes. Is this right? Should we explicitly specify `clang-11`, `clang-10`, and `clang`?

Changed to  
`find_program(CMAKE_C_COMPILER clang-11 clang-10 clang)`

File: tests/switchless/switchless_test.edl [hunk 2]
```diff
@@ -42,7 +42,7 @@ enclave {
             [string, in] const char* str1,
             [in] char str2[100]);
 
-	public void enc_test_large_switchless_ocall();
+	    public void enc_test_large_switchless_ocall();
```
[the remark this answers is not in the document]
Comment (asvrada) [reply tests/switchless/switchless_test.edl:45]: I think previously the indentation is missing for this method, so I added it along with this PR

**Pull request merged**